src/transaction/components/TransactionProvider.test.tsx

```diff
@@ -382,34 +365,6 @@ describe('TransactionProvider', () => {
       );
     });
   });
-
-  it('should call onSuccess when receiptArray has receipts', async () => {
```

Comment:
Same for this one. I am rewriting the entire logic in a more simple way so we can actually test this.